modules/apigee_edge_teams/src/Form/TeamMembersFormBase.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge_teams\Form;
+
+use Drupal\apigee_edge_teams\Entity\TeamRoleInterface;
+use Drupal\Core\Form\FormBase;
+
+/**
+ * Base team members form.
+ */
+abstract class TeamMembersFormBase extends FormBase {
+
+  /**
+   * The team from the route.
+   *
+   * @var \Drupal\apigee_edge_teams\Entity\TeamInterface
+   */
+  protected $team;
+
+  /**
+   * Team role storage.
+   *
+   * @var \Drupal\apigee_edge_teams\Entity\Storage\TeamRoleStorageInterface
+   */
+  protected $teamRoleStorage;
+
+  /**
+   * Team member role storage.
+   *
+   * @var \Drupal\apigee_edge_teams\Entity\Storage\TeamMemberRoleStorage
+   */
+  protected $teamMemberRoleStorage;
+
+  /**
+   * Returns an array of team role options keyed by team role id.
+   *
+   * @return array
+   *   Role options.
+   */
+  protected function getRoleOptions() {
+    $role_options = array_reduce($this->teamRoleStorage->loadMultiple(), function (array $carry, TeamRoleInterface $role) {
```

Comment:
We should probably add a constructor here to avoid cases where `$this->teamRoleStorage` might be null. Or add an `abstract function getTeamRoleStorage(): TeamRoleStorageInterface;`.

Reply:
@arshad I moved __construct() and create() to the base form to address this. Good catch :+1: